git_externals/git_externals.py

```diff
@@ -396,39 +396,43 @@ def autosvn_update_checkout(reset):
                 gitext_up(recursive, entries, reset=reset, use_gitsvn=use_gitsvn)
 
 
-def gitext_recursive_info(root_dir):
-    git_exts = load_gitexts()
-    git_exts = {ext_repo: ext for ext_repo, ext in git_exts.items()
+def gitext_recursive_info(root_dir, recursive=True, externals=[]):
+    git_exts = {ext_repo: ext for ext_repo, ext in load_gitexts().items()
                 if os.path.exists(os.path.join(externals_root_path(), get_repo_name(ext_repo)))}
 
     for ext_repo, ext in git_exts.items():
-        print_gitext_info(ext_repo, ext, root_dir=root_dir)
-
-    for ext_repo, ext in git_exts.items():
         entries = [os.path.realpath(d)
                    for t in git_exts[ext_repo]['targets'].values()
                    for d in t]
 
         cwd = os.getcwd()
+        repo_name = get_repo_name(ext_repo)
+        if externals and repo_name not in externals:
+            continue
 
-        with chdir(os.path.join(externals_root_path(), get_repo_name(ext_repo))):
+        with chdir(os.path.join(externals_root_path(), repo_name)):
             filtered = filter_externals_not_needed(load_gitexts(), entries)
+            print_gitext_info(ext_repo, ext, root_dir, checkout=os.getcwd())
 
-            for dsts in ext['targets'].values():
-                for dst in dsts:
-                    real_dst = os.path.realpath(os.path.join(cwd, dst))
+            # if required, recurse into the externals repo of current external
+            if recursive:
+                for dsts in ext['targets'].values():
+                    for dst in dsts:
+                        real_dst = os.path.realpath(os.path.join(cwd, dst))
 
-                    has_deps = any([os.path.realpath(d).startswith(real_dst)
-                                    for e in filtered.values()
-                                    for ds in e['targets'].values()
-                                    for d in ds])
+                        has_deps = any([os.path.realpath(d).startswith(real_dst)
```

Comment:
I didn't touch this code, just added the comment saying that that here starts the recursion into nested externals